src/basic/strxcpyx.c

```diff
@@ -93,7 +93,7 @@ size_t strscpyl(char *dest, size_t size, const char *src, ...) {
         do {
                 size = strpcpy(&s, size, src);
                 src = va_arg(va, char *);
-        } while (src != NULL);
+        } while (src != NULL && size > 0);
```

Comment:
I think this would be "premature optimization". E.g. those functions cannot do anything when `size == 1` either, so you could just as well stop the loop then. The reason I think this isn't useful is that the argument list can only be of a finite and small length, and overflow is the exceptional case. So it's better to keep the code simple.